language/tools/resource-viewer/src/lib.rs

```diff
@@ -34,7 +34,7 @@ mod resolver;
 #[derive(Debug)]
 pub struct AnnotatedAccountStateBlob(BTreeMap<StructTag, AnnotatedMoveStruct>);
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct AnnotatedMoveStruct {
```

Comment:
The dynamic analysis uses `AnnotatedMoveValue` to extract addresses from published resources. This required some small tweaks (allowing `clone`, making it possible to extract the type).

Reply:
I see. I didn't really expect `AnnotatedMoveValue ` to be used in a production setup. Can we use `MoveValue` instead? IMHO the `AnnotatedMoveValue` buys us only those extra field name info that we don't need here?

Reply:
I looked into this and although I agree this is the right longterm direction, using `MoveValue` here is going to be a lot of work. The logic for getting a `MoveValue` from bytes is very entangled with the internals of the VM loader and cache + not at all easy to use from the outside. I believe @metajack also encountered this problem when working on the sqlizer.

Because this is an early prototype of the read/write set analysis that already uses components we'd prefer not to use in production (e.g., prover stackless IR + `GlobalEnv` instead of working directly over bytecode), I would propose that we go forward with `AnnotatedMoveValue` for now. Our initial goal should be to evaluate the long-term feasibility of using the read/write set analysis at runtime in the parallel execution component; we all agree that the analysis is overkill for the current, simple set of tx scripts we will have for the foreseeable future. If the outlook for the analysis looks good, we can undertake the fairly large project of making it production-ready from a dependencies, performance, and testing standpoint.

Reply:
Sounds good. In the worst case we can just statically cache the inferred result in the code so that we only need to support a minimal list of scripts.